src/components/Menu/__stories__/ChildrenProps.tsx

```diff
@@ -0,0 +1,79 @@
+import { fireEvent, screen } from '@storybook/testing-library'
+import Menu from '..'
+import { Template } from './Template'
+
+export const ChildrenProps = Template.bind({})
+
+const customCodeSnippet = `<Menu disclosure={<Touchable name="menu" title="menu"><Icon color="neutral" name="dots-horizontal" size={24}/></Touchable>}>
+{({ toggle }) => (
+  <>
+    <Menu.Item>default</Menu.Item>
+    <Menu.Item onClick={toggle} variant="danger">
+      Danger
+    </Menu.Item>
+    <Menu.Item onClick={toggle} variant="nav">
+      Nav
+    </Menu.Item>
+    <Menu.Item onClick={toggle} variant="nav" disabled>
+      Nav disabled
+    </Menu.Item>
+    <Menu.Item onClick={toggle} variant="nav" disabled borderless>
+      Nav disabled borderless
+    </Menu.Item>
+    <Menu.Item
+      href="/?path=/docs/components-navigation-menu--children-props"
+      onClick={toggle}
+    >
+      Menu Item Link
+    </Menu.Item>
+  </>
+)}
+</Menu>`
```

Comment:
As you can see, we need to manually overwrite the code snippet for custom cases like this one.

(In that case that's because passing a function to `children` param isn't working as expected)